### PR TITLE
MDX: Handle quotes / template literals in title

### PR DESCRIPTION
**Pull request by atanasster**
Issue: ability to have string templates and expressions in Meta title

## What I did
* in mdx-compiler, if node is not StringLiteral, generate code for the title and leave it to CSF compiler to evaluate

## How to test

Added story to official-storybook and added snapshot testing

### CODE REVIEW

File: addons/docs/src/mdx/mdx-compiler-plugin.js [hunk 1]
```diff
@@ -133,7 +133,20 @@ function genMeta(ast) {
...
+    if (title.type === 'StringLiteral') {
+      title = "'".concat(title.value, "'");
+    } else {
+      try {
+        // generate code, so the expression is evaluated by the CSF compiler
+        const { code } = generate(title, {});
+        // remove the curly brackes at start and end of code
+        title = code.replace(/^\{(.+)\}$/, '$1');
+      } catch (e) {
+        // eat exception if title parsing didnt go well
```
Comment (shilman) [inline addons/docs/src/mdx/mdx-compiler-plugin.js]: Maybe log it?

**Comment by atanasster** [comment]
this PR now includes also #9068

File: addons/docs/src/mdx/mdx-compiler-plugin.js (hunk 1 above)
Comment (atanasster) [reply addons/docs/src/mdx/mdx-compiler-plugin.js]: Thanks, I added a console.warn now

**Review by shilman — APPROVED** [review]
LOVE IT ❤️ 

Thanks for bearing with me on the exception 😀

**Pull request merged**